code/modules/mob/living/simple_animal/hostile/zombie.dm

```diff
@@ -55,11 +55,13 @@
 
 	if(istype(target, /obj/machinery/door/airlock))
 		if(!removingairlock)
-			src << "<span class='notice'>You start tearing apart the airlock...</span>"
-			playsound(src.loc, 'sound/hallucinations/growl3.ogg', 50, 1)
 			var/obj/machinery/door/airlock/A = target
 			removingairlock = 1
-			if(do_after(src, 200, 0, A, 1))
+			src << "<span class='notice'>You start tearing apart the airlock...</span>"
+			playsound(src.loc, 'sound/machines/airlock_alien_prying.ogg', 100, 1)
+			sleep(20)
```

Comment:
do_after will work as long as you stay in one stop once it starts, regardless of where that spot actually IS.
So yes, you do need the checks, or to remove this